### ISSUE TITLE
Add Linux backend

### ISSUE THREAD

**Issue by ThomCoder** [ThomCoder/Install_A_Bot#5]
~~Blocked by #3~~

**Comment by ThomCoder** [comment]
this issue now has a special role.
A Docker container will be used as a testbench for the Linux logic and general backend functionality. This not only helps the developers' host systems clean, it will also make development more platform independent.

**Comment by ThomCoder** [comment]
As of commit 53ecb0b there is very dumb initial support for installing packages via a package manager (e.g. "apt install"). This however only works on "real" Linux machines. Something seems to be wrong with the attached docker_* scripts.

One drawback though is that the "Install_A_Bot" binary has to run as root in order to use package managers non interactively.

All of this will be finetuned (ideally with a working docker container).

**Comment by theswiftfox** [comment]
i think there might be a crate to make asking vor privileges trivial. could be worth investigating :)

**Comment by ThomCoder** [comment]
> i think there might be a crate to make asking vor privileges trivial. could be worth investigating :)

Good idea, I'll look into it. From a quick glance I found [runas](http://mitsuhiko.github.io/rust-runas/runas/). Maybe something along the lines of that will do the trick.

**Comment by ThomCoder** [comment]
At the current state we have some very basic (i.e. dumb) Linux support. While it basically works the module still needs some love. I'm on it.

**Comment by theswiftfox** [comment]
Sounds great!
If I can assist in any way let me know :)
